### ISSUE TITLE
Blocks of type "route" are not expected here

### ISSUE THREAD

**Issue by sgript** [hashicorp/terraform#21228]
Hi there,

Recently we upgraded from 0.11 to 0.12 Beta 1. A member of my team has the version:
```Terraform v0.12.0-beta1
+ provider.aws v1.60.0-dev20190216h00-dev
```

And when he runs his execution plan, all seems to work. However, when I run the same, I get the error:
```
Error: Unsupported block type

  on ../../modules/aws/route_table/main.tf line 4, in resource "aws_route_table" "route_table":
   4:   route {

Blocks of type "route" are not expected here. Did you mean to define argument
"route"? If so, use the equals sign to assign it a value.
```
And I am running the versions:
```
Terraform v0.12.0-beta1
+ provider.aws v2.9.0
```

However, I have tried to pin my provider block to use the version we have had success with but it's not being found (perhaps I set it incorrectly?). I've also tried to pin it to 1.60.0  (non dev), but it then tells me:
```
Provider version 2.7.0 is the earliest compatible version. Select it with
the following version constraint:

	version = "~> 2.7"
```

In relation to the actual error we're seeing (see above - "route" not being expected), here's the problematic code:
```
resource "aws_route_table" "route_table" {
  vpc_id = "${var.vpc_id}"

  route {
    cidr_block = "${var.cidr_block}"
    gateway_id = "${var.gateway_id}"
  }

  tags = {
    Name = "${var.environment}-${uuid()}"
  }
}
```

This is called twice:

```
module "route-table" {
  source ="../../aws/route_table"
  environment = "${var.environment}"

  vpc_id = "${module.vpc.id}"
  cidr_block = "0.0.0.0/0"
  gateway_id = "${module.internet_gateway.id}"
}

module "route-table-nat" {
  source ="../../aws/route_table"
  environment = "${var.environment}"

  vpc_id = "${module.vpc.id}"
  cidr_block = "0.0.0.0/0"
  gateway_id = "${module.nat_gateway.id}"
}
```

I don't see the issue here, could someone shed some light into what this might be?

Thanks.

**Comment by apparentlymart** [comment]
Hi @sgript! Sorry for the confusing behavior here.

I think this is because you are using beta 1 rather than beta 2. Between the two we had to make some changes to the representation of provider schemas in order to fix some behavior regressions, and the Core beta 1 release doesn't know how to understand those changes and so it is misinterpreting the schema returned by the provider.

You also should not need to use a dev snapsnot for the `aws` provider at this point, because versions 2.7.0, 2.8.0, and 2.9.0 of the provider are natively 0.12-compatible and can be installed automatically with `terraform init`. The 2.9.0 release is newer than the dev snapshot build you mentioned here and so has some additional fixes to improve Terraform 0.12 compatibility.

**Comment by sgript** [comment]
Hey @apparentlymart 

Thanks for the response! Since this reply, I upgraded to 0.12.0-rc1 and I'm having a different issue, albeit still with route53 records. See below:

```
2019-05-09T09:31:32.879+0100 [DEBUG] plugin.terraform-provider-aws_v2.9.0_x4: 2019/05/02019/05/09 09:31:32 [ERROR] module.staging-infrastructure.module.cassie.module.subdomain: eval: *terraform.EvalReadState, err: unsupported attribute "failover"
2019/05/09 09:31:32 [ERROR] module.staging-infrastructure.module.cassie.module.subdomain: eval: *terraform.EvalSequence, err: unsupported attribute "failover"
2019/05/09 09:31:32 [ERROR] module.staging-infrastructure.module.csi.module.subdomain: eval: *terraform.EvalReadState, err: unsupported attribute "failover"
2019/05/09 09:31:32 [ERROR] module.staging-infrastructure.module.csi.module.subdomain: eval: *terraform.EvalSequence, err: unsupported attribute "failover"

Error: unsupported attribute "failover"



Error: unsupported attribute "failover"



Error: unsupported attribute "failover"



Error: unsupported attribute "failover"



Error: unsupported attribute "failover"



Error: unsupported attribute "failover"
```

The trace leads to:
```
module "subdomain" {
  source      = "../../aws/route53_record"
  environment = "${var.environment}"

  veygo_domain = "${var.veygo_domain}"
  name         = "${var.subdomain}"
  records      = "${module.eb.cname}"
}
```

Calling the route53 record module, which is written like:
```
resource "aws_route53_record" "record" {
  zone_id = "${data.aws_route53_zone.zone.zone_id}"
  name    = "${var.name}"
  type    = "${var.type}"
  records = ["${var.records}"]
  ttl     = 60
}
```

Any advice would be really appreciated. 🙂

**Comment by sgript** [comment]
Update: We found the fix for this. The state needed to be updated to drop attributes no longer supported, such as failover. See [this](https://github.com/terraform-providers/terraform-provider-aws/issues/8586).

**Comment by apparentlymart** [comment]
Hi @sgript!

Since the original problem you reported seems to be fixed, I'm going to close this out. I believe the other issue you raised here is also fixed, but that was tracked in some other GitHub issues which I don't have to hand right now. Some of these fixes landed after the initial 0.12-compatible AWS provider releases, so it may be necessary to upgrade both the AWS provider _and_ Terraform CLI to latest in order to see the fixes.

Thanks for following up, and sorry for the delayed response.
